src/BioFSharp.Vis/Upset.fs

```diff
@@ -0,0 +1,201 @@
+        let setIDsPresent, setIDsNotPresent =
+            labelIDs
+            |> Array.partition (fun (label,_) -> intersectingSets |> List.contains label)
+        let lineChart =
+            let idWithPosition =
+                setIDsPresent
+                |> Array.map (fun (_,id) ->
+                    position,id
+                )
+            createIntersectionLineChart idWithPosition markerSize colorIntersecting
+        let pointChart =
+            let idWithPosition =
+                setIDsNotPresent
+                |> Array.map (fun (_,id) ->
+                    position,id
+                )
+            createIntersectionPointChart idWithPosition markerSize colorNotIntersecting
+        [
+            pointChart
+            lineChart
+        ]
+        |> Chart.combine
+
+    /// Creates a bar chart with the set sizes
+    let createSetSizePlot (labels:array<string>) (sets:array<Set<'a>>) (maxY: float) (color: Color) (domainSet: float*float) (textFont: Font) =
+        let labelCount =
+            Array.map2 (fun label (set: Set<'a>) ->
+                label, set.Count
+            ) labels sets
+        let maxSetSize =
+            labelCount
+            |> Array.maxBy snd
+            |> snd
+        Chart.Bar (labelCount, Color = color)
+        |> Chart.withXAxisStyle("Set Size",MinMax=(float maxSetSize,0.), Domain=domainSet, TitleFont=textFont)
+        |> Chart.withYAxis (createLinearAxisWithRange maxY)
+        |> Chart.withTraceName(ShowLegend=false)
+
+    /// Creates a bar chart with the intersection sizes
+    let createIntersectionSizePlots (intersectionCount: (string list*int)[]) (maxX: float) (color: Color) (domainIntersection: float*float) (textFont: Font) =
+        intersectionCount
+        |> Array.map snd
+        |> fun count -> 
+            Chart.Column(count, Color = color)
+        |> Chart.withXAxis (createLinearAxisWithRangeDomain maxX domainIntersection)
+        |> Chart.withYAxisStyle("Intersection Size", TitleFont=textFont)
+        |> Chart.withTraceName(ShowLegend=false)
+
+    /// Aligns the map with Setelement->Feature to the intersections
+    let alignIntersectionData (venn: (string list * Set<'a>)[]) (setData: Map<'a,'b>) =
+        venn
+        |> Array.map (fun (_,set) ->
+            set
+            |> Set.toArray
+            |> Array.map (fun entry ->
+                setData.[entry]
+            )
+        )
+
+    /// Creates the intersection feature plot with the given charting function
+    let createIntersectionDataPlots (intersectionData: array<array<'b>>) (title: string) (maxX: float) (chartFun: array<'b> -> GenericChart.GenericChart) (domainIntersection: float*float) (textFont: Font) =
+        intersectionData
+        |> Array.map chartFun
+        |> Chart.combine
+        |> Chart.withLegend false
+        |> Chart.withXAxis (createLinearAxisWithRangeDomain maxX domainIntersection)
+        |> Chart.withYAxisStyle (title, TitleFont=textFont)
+
+module Upset =
```

Comment:
Please refactor this as overload for the `Chart` class. Example can be seen [here](https://github.com/fslaborg/fslaborg.github.io/blob/a92638214e3873551a522cc2bea59fb985cc6f1b/src/content/tutorials_src/008_sequence_features.fsx#L519). This has the additional advantage that you can use optional arguments.